gvr-controller/app/src/main/java/org/gearvrf/sample/controller/SampleMain.java

```diff
@@ -173,162 +184,241 @@ public void onInit(GVRContext gvrContext) {
         attachBoundsCollider(object);
         mainScene.addSceneObject(object);
 
-
         GVRAssetLoader assetLoader = gvrContext.getAssetLoader();
-
-        GVRInputManager inputManager = gvrContext.getInputManager();
-        inputManager.addCursorControllerListener(cursorControllerListener);
-        for (GVRCursorController cursor : inputManager.getCursorControllers()) {
-            cursorControllerListener.onCursorControllerAdded(cursor);
-        }
-
-        Future<GVRTexture> futureTexture = assetLoader.loadFutureTexture(new
-                GVRAndroidResource(gvrContext, R.drawable.skybox_gridroom));
+        Future<GVRTexture> futureTexture = assetLoader.loadFutureTexture(
+                new GVRAndroidResource(gvrContext, R.drawable.skybox_gridroom));
         GVRMaterial material = new GVRMaterial(gvrContext);
         GVRSphereSceneObject skyBox = new GVRSphereSceneObject(gvrContext, false, material);
         skyBox.getTransform().setScale(SCALE, SCALE, SCALE);
         skyBox.getRenderData().getMaterial().setMainTexture(futureTexture);
         mainScene.addSceneObject(skyBox);
     }
 
-    private CursorControllerListener cursorControllerListener = new CursorControllerListener() {
+    private CursorControllerListener cursorControllerListener = new CursorControllerListener()
+    {
         @Override
-        public void onCursorControllerAdded(GVRCursorController gvrCursorController) {
-
-            if (gvrCursorController.getControllerType() == GVRControllerType.CONTROLLER) {
-                android.util.Log.d(TAG, "Got the orientation remote controller");
-                GVRSceneObject cursor = new GVRSceneObject(mGVRContext,
-                        mGVRContext.createQuad(1f, 1f),
-                        mGVRContext.getAssetLoader().loadTexture(new GVRAndroidResource(mGVRContext, R.raw.cursor)));
-                cursor.getRenderData().setDepthTest(false);
-                cursor.getRenderData().setRenderingOrder(100000);
-                controller = new GVRGearControllerSceneObject(mGVRContext);
-                controller.setCursorController(gvrCursorController);
-                mPicker.setPickRay(0.0f, 0.0f, 0.0f, 0.0f, 0.0f, DEPTH);
-                controller.setRayDepth(DEPTH);
-                controller.setCursor(cursor);
-                controller.attachComponent(mPicker);
-                gvrCursorController.addControllerEventListener(controllerEventListener);
-            }else{
+        public void onCursorControllerAdded(GVRCursorController gvrCursorController)
```

Comment:
The old api was already making users do too much. On the surface at least, it now looks the users have even more work to do. Can't all these ifs be pushed into the framework?

Reply:
The 3D Cursor extension attempts to solve this problem. That is our current method of pushing these ifs into the framework. It supports configuration files and the idea of multiple active cursors.

Reply:
gvr-controller does not use the 3dcursorlibrary. All I can say that now there is even less abstraction than before, without the (flawed) 3dcursorlibrary..